src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -614,4 +592,27 @@ export default class MongooseAdapter {
 
     return words.join(" ");
   }
+
+  static makeIdQuery(idOrIds) {
+    const idIsValid = mongoose.Types.ObjectId.isValid;
+    let idQuery;
+
+    if (typeof idOrIds === "string") {
+      if (!idIsValid(idOrIds)) {
+        throw new APIError(404, undefined, "No matching resource found.");
+      }
+
+      idQuery = idOrIds;
+    } else if (Array.isArray(idOrIds)) {
+      let validIds = idOrIds.filter(idIsValid);
```

Comment:
It's true that, right now, multiple ids in a request usually come in only via a `labelMapper`, and those should only be returning ids that are valid. But a request like `/collection?filter[simple][_id][$in]=id1,id2,id3` is also possible (and will continue to be supported even after removing the nasty mongo-specific operators). In such a case, if one of the ids is invalid, I think the correct thing to do is to 400 the entire request, right?